Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/sdk/common/SharedKeyHttpObservable.java

```diff
@@ -21,40 +21,106 @@
  */
 package com.microsoft.azure.hdinsight.sdk.common;
 
+import com.microsoft.azure.hdinsight.sdk.rest.ObjectConvertUtils;
+import com.microsoft.azure.hdinsight.sdk.storage.adlsgen2.AccessKeyList;
 import com.microsoft.azure.hdinsight.sdk.storage.adlsgen2.SharedKeyCredential;
+import com.microsoft.azure.management.Azure;
 import com.microsoft.azure.storage.core.Utility;
+import com.microsoft.azuretools.authmanage.AuthMethodManager;
+import com.microsoft.azuretools.authmanage.models.SubscriptionDetail;
 import com.microsoft.azuretools.azurecommons.helpers.NotNull;
+import com.microsoft.azuretools.azurecommons.helpers.Nullable;
 import org.apache.http.Header;
 import org.apache.http.NameValuePair;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpEntityEnclosingRequestBase;
+import org.apache.http.client.methods.HttpPost;
 import org.apache.http.client.methods.HttpRequestBase;
 import org.apache.http.message.BasicHeader;
 import org.apache.http.message.HeaderGroup;
+import rx.Observable;
+import rx.exceptions.Exceptions;
 
-import java.security.InvalidKeyException;
+import java.io.IOException;
+import java.net.UnknownServiceException;
 import java.util.ArrayList;
-import java.util.Arrays;
 import java.util.List;
+import java.util.Optional;
 import java.util.UUID;
 
-public class SharedKeyHttpObservable extends HttpObservable {
+public class SharedKeyHttpObservable extends AzureHttpObservable {
```

Comment:
>AzureHttpObservable [](start = 45, length = 19)

Please do not inherit `AzureHttpObservable`, which is for OAuth authentication.

Reply:
req to get access key needs aad token

---
In reply to: [271165829](https://github.com/Microsoft/azure-tools-for-java/pull/2961#discussion_r271165829) [](ancestors = 271165829)